static/js/message_list_view.js

```diff
@@ -326,8 +327,35 @@ export class MessageListView {
 
         message_container.is_hidden = is_hidden;
         // Make sure the right thing happens if the message was edited to mention us.
-        message_container.contains_mention = message_container.msg.mentioned && !is_hidden;
+        if (!is_hidden && message_container.msg.mentioned) {
+            // Currently the API does not differentiate between a group mention and
+            // a user mention. For now, we parse the markdown to see if the message
+            // mentions the user.
+            let is_user_mention = false;
+            const $msg = $(message_container.msg.content);
+            $msg.find(".user-mention:not(.silent)").each(function () {
```

Comment:
We should try to do this without jQuery to make this faster.

Reply:
That sounds great! Do you know what that would look like?